### PR TITLE
Add MPAS-Analysis parameters for mapping in parallel

### PR DESCRIPTION
**Pull request by xylar**
The new E3SM-Unified supports ESMF map-file generation in parallel with system MPI on most machines.  (ESMF could not be built with system MPI on Cori.)  On Anvil and Chrysalis, MPAS-Analysis will not be able to run unless ESMF calls are made with `srun` (which is not the MPAS-Analysis default).

This merge updates the config options related to parallel map-file generation to be consistent with the new MPAS-Analysis and E3SM-Unified.

### CODE REVIEW

File: zppy/templates/mpas_analysis.bash
Comment (xylar) [inline zppy/templates/mpas_analysis.bash:146]: This is the MPAS-Analysis default but we might want to make this a zppy config option, too, for big meshes.

**Comment by xylar** [comment]
@golaz and @forsyth2, this is needed for `zppy` to work with MPAS-Analysis in E3SM-Unified 1.5.0.  Once this is merged (along with any other changes you want to make now), we will need to build `zppy` 1.1.0rc2 and I will need to deploy E3SM-Unified 1.5.0rc3 with this new version.

**Comment by xylar** [comment]
I switched this to a draft because I made some progress today on both Cori-Haswell and Compy, and I need to update this PR accordingly once I release MPAS-Analysis 1.4.0rc3.

**Comment by xylar** [comment]
## Testing

I ran MPAS-Analysis via zppy on the 20210305.v2beta3GM900.piControl.ne30pg2_EC30to60E2r2.compy simulation on Compy using this branch and E3SM-Unified 1.5.0rc3.

After a few false starts, it's working well!

<details>
 <summary>Compy config file</summary>

``` cfg

[default]
input = /compyfs/vanr302/E3SM_simulations/20210305.v2beta3GM900.piControl.ne30pg2_EC30to60E2r2.compy
output = /compyfs/asay932/analysis/zppy_test/20210305.v2beta3GM900.piControl.ne30pg2_EC30to60E2r2.compy
case = 20210305.v2beta3GM900.piControl.ne30pg2_EC30to60E2r2.compy
www = /compyfs/www/asay932/analysis/zppy_test/20210305.v2beta3GM900.piControl.ne30pg2_EC30to60E2r2.compy
partition = slurm
environment_commands = source /share/apps/E3SM/conda_envs/test_e3sm_unified_1.5.0rc3_compy.sh

[climo]
active = False

[ts]
active = False

[e3sm_diags]
active = False

[e3sm_diags_vs_model]
active = False

[amwg]
active = False

[mpas_analysis]
active = True
shortTermArchive = True
walltime = "4:00:00"
parallelTaskCount = 6
mapMpiTasks = 6
ts_years = "1-50",
enso_years = "11-50",
climo_years ="21-50",
mesh = "EC30to60E2r2"

[global_time_series]
active = False
```

</details>

**Review by golaz — APPROVED** [review]

File: zppy/templates/mpas_analysis.bash
Comment (forsyth2) [inline zppy/templates/mpas_analysis.bash:157]: Looks like you meant to have this be three lines of comments?

**Review by forsyth2 — APPROVED** [review]
I haven't tested myself, but code changes look good. Just one minor comment on on comment formatting.

File: zppy/templates/mpas_analysis.bash
Comment (xylar) [reply zppy/templates/mpas_analysis.bash:157]:
```suggestion
# "None" if ESMF should perform remapping in serial without a command, or one of
# "srun" or "mpirun" if it should be run in parallel  (or in serial but with a
# command)
```

Comment (xylar) [reply zppy/templates/mpas_analysis.bash]: @forsyth2, Good catch!  I messed this one up in the terminal window and copied it a bunch of places, including here, before I noticed.

**Comment by xylar** [comment]
@forsyth2 and @golaz, please feel free to merge this (and any of the other PRs) when you're ready and to make a new `rc2` tag for me to work with.

**Comment by forsyth2** [comment]
> @forsyth2 and @golaz, please feel free to merge this (and any of the other PRs) when you're ready and to make a new `rc2` tag for me to work with.

@xylar Chris and I have both approved this PR and #65, so I can merge these two. I can also merge #68 since I see Chris approved. I notice #64 is still a draft -- do you want me to review that and merge it too before we create the `rc2` tag?

**Comment by xylar** [comment]
> I notice #64 is still a draft -- do you want me to review that and merge it too before we create the rc2 tag?

Sorry, I meant to explain that.  #64 is based off of this PR so if you merge #64 now, it will also merge this PR.  You can review it now and I will take it out of draft mode.  Just be careful to merge this PR first and then #64.

**Comment by forsyth2** [comment]
@xylar Sounds good, I can merge this PR, #64, #65, and #68. Do you want to squash your commits into one for this PR and #64?

**Comment by xylar** [comment]
@forsyth2, okay, I have rebased and squashed.  Should be good to merge.  I can't take care of #64 until this is merged.

**Pull request merged**
